homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -28,56 +77,157 @@ async def async_setup_entry(
     """Set up a Waze travel time sensor entry."""
     name = config_entry.data.get(CONF_NAME, DEFAULT_NAME)
     coordinator = config_entry.runtime_data
+    entry_id = config_entry.entry_id
+    options = config_entry.options
 
-    sensor = WazeTravelTimeSensor(config_entry.entry_id, name, coordinator)
-
-    async_add_entities([sensor], False)
+    sensors: list[WazeTravelTimeSensor] = [
+        WazeTravelTimeSensor(
+            entry_id,
+            name,
+            sensor_description,
+            coordinator,
+        )
+        for sensor_description in sensor_descriptions(options[CONF_VEHICLE_TYPE])
+    ]
+    sensors.append(
+        DurationSensor(entry_id, name, coordinator, options[CONF_VEHICLE_TYPE])
+    )
+    sensors.append(OriginSensor(entry_id, name, coordinator))
+    sensors.append(DestinationSensor(entry_id, name, coordinator))
+    async_add_entities(sensors, False)
 
 
 class WazeTravelTimeSensor(CoordinatorEntity[WazeTravelTimeCoordinator], SensorEntity):
     """Representation of a Waze travel time sensor."""
 
     _attr_attribution = "Powered by Waze"
-    _attr_native_unit_of_measurement = UnitOfTime.MINUTES
-    _attr_suggested_display_precision = 0
-    _attr_device_class = SensorDeviceClass.DURATION
-    _attr_state_class = SensorStateClass.MEASUREMENT
-    _attr_device_info = DeviceInfo(
-        entry_type=DeviceEntryType.SERVICE,
-        name="Waze",
-        identifiers={(DOMAIN, DOMAIN)},
-        configuration_url="https://www.waze.com",
-    )
-    _attr_translation_key = "waze_travel_time"
+    _attr_has_entity_name = True
+    entity_description: WazeSensorDescription
 
     def __init__(
         self,
-        unique_id: str,
+        unique_id_prefix: str,
         name: str,
+        description: WazeSensorDescription,
         coordinator: WazeTravelTimeCoordinator,
     ) -> None:
         """Initialize the Waze travel time sensor."""
         super().__init__(coordinator)
-        self._attr_unique_id = unique_id
-        self._attr_name = name
+        self.entity_description = description
+        self._attr_unique_id = (
+            unique_id_prefix
+            if description.key == ATTR_DURATION
+            else f"{unique_id_prefix}_{description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            name=name,
+            identifiers={(DOMAIN, DOMAIN)},
```

Comment:
```suggestion
            identifiers={(DOMAIN, name)},
```
As far as I understand this should be a unique tuple by which a device can be identified. Since the name should be unique it might be a good fit. Otherwise `config_entry.entry_id`